apps/web/lib/services/audit-logger.ts

```
@@ -0,0 +1,111 @@
+			metadata: params.metadata,
+			errorCode: params.errorCode,
+			durationMs: params.durationMs,
+		}
+
+		// Always emit structured JSON to console
+		console.info('[AUDIT]', JSON.stringify(entry))
+
+		try {
+			const supabase = createSupabaseBrowserClient()
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- audit_logs not in generated types
+			const { error } = await (supabase as any)
```

Biome (2.4.7)

[error] Unexpected any. Specify a different type.

(lint/suspicious/noExplicitAny)